contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,271 @@
+            [context, final_continuation, pChecker] (Status status,
+                                           std::string &&body) {
+              std::string ruleset_id;
+              if (status.ok()) {
+                pChecker->env_->LogDebug(
+                    std::string("GetReleasName succeeded with ") + body);
+                std::tie(status, ruleset_id) = pChecker->ParseReleaseResponse(&body);
+              } else {
+                pChecker->env_->LogError(std::string("GetReleaseName for ")
+                                         + pChecker->GetReleaseUrl(context)
+                                         + " with status " + status.ToString());
+                status = Status(Code::INTERNAL, kFailedFirebaseReleaseFetch);
+              }
 
-  // Get Auth token for accessing Firebase Rules service.
-  const std::string &GetAuthToken();
+              // If the parsing of the release body is successful, then call the
+              // Test Api for firebase rules service.
+              if (status.ok()) {
+                pChecker->CheckAuth(ruleset_id, context, final_continuation);
+              } else {
+                final_continuation(status);
+              }
+            });
+}
 
-  // Request context.
-  std::shared_ptr<context::RequestContext> context_;
+void AuthzChecker::CheckAuth(
+    std::string ruleset_id,
+    std::shared_ptr<context::RequestContext> context,
+    std::function<void(Status status)> continuation) {
+  auto pChecker = GetPtr();
 
-  // Pointer to access ESP running environment.
-  ApiManagerEnvInterface *env_;
+  HttpFetch(std::string(kFirebaseServerStaging) + "v1/" + ruleset_id +
+            ":test?alt=json",
+            std::string("POST"), BuildTestRequestBody(context),
+            [context, continuation, pChecker, ruleset_id]
+            (Status status, std::string &&body) {
 
-  // The final continuation function.
-  std::function<void(Status status)> on_done_;
-};
+              if (status.ok()) {
+                pChecker->env_->LogDebug(
+                    std::string("Test API succeeded with ") + body);
+                status = pChecker->ParseTestResponse(context, &body);
+              } else {
+                pChecker->env_->LogError(std::string("Test API failed with ")
+                    + status.ToString());
+                status = Status(Code::INTERNAL, kFailedFirebaseTest);
+              }
 
-AuthzChecker::AuthzChecker(std::shared_ptr<context::RequestContext> context,
-                           std::function<void(Status status)> continuation)
-    : context_(context),
-      env_(context_->service_context()->env()),
-      on_done_(continuation) {}
+              continuation(status);
+            });
+}
 
-void AuthzChecker::Check() {
-  // TODO: Check service config to see if "useSecurityRules" is specified.
-  // If so, call Firebase Rules service TestRuleset API.
+std::pair<Status, std::string> AuthzChecker::ParseReleaseResponse(
```

Comment:
can we pass out ruleset_id in the argument. e.g.
Status Parse(...,   string* ruleset_id)